src/hooks/usePagination.js

```diff
@@ -37,16 +37,6 @@ export const usePagination = props => {
     ]
   } = props
 
-  useLayoutEffect(() => {
-    setState(
-      old => ({
-        ...old,
-        pageIndex: 0
-      }),
-      actions.pageChange
-    )
-  }, [rows, filters, groupBy, sortBy])
```

Comment:
When rows, filters, groupBy or sortBy are changed, it essentially invalidates the data on pages that you've previously viewed (say you were on page 2, page 1's data has likely changed now). This effect is there to take you back to page 1, so you can see the new data that was generated from your data + filters + grouping + sorting.

I fully intend to keep this as the default (since this is what I have found to be the default in almost every other table), but I would also immediately accept a PR to add an option to disable this effect if you choose to do so.

Reply:
And I'll admit that that is a feature that I want and makes sense.  My problem is that with remote data it simply resets to page zero on every refresh.  You can see that on the codesandbox if you switch from the local react-table files and use the alpha-5 version, the serverData version can no longer page.

Reply:
actually, should rows be in that dependency list?  In a scenario where the remote data is being updated periodically, but everything else is remaining constant, would you want it to jump back to the first page?

Reply:
I think you just proved the need for this option. Can't wait to see the PR ;)

Reply:
And yes, I think by default, rows should be in there. 99% of the time if the core data changes, I think the page should reset. There are clearly times when that shouldn't happen though. A boolean should be enough for now.